geomstats/stratified_geometry/wald_space.py

```diff
@@ -0,0 +1,1024 @@
+            """Raise an exception when less equal is not true."""
+
+        x_parts = [set(x) for x in self.partition]
+        y_parts = [set(y) for y in other.partition]
+        # (i)
+        try:
+            cover = {
+                i: [j for j, y in enumerate(y_parts) if x.issubset(y)][0]
+                for i, x in enumerate(x_parts)
+            }
+        except IndexError:
+            return False
+        # (ii)
+        try:
+            for (i, j), x in zip(cover.items(), x_parts):
+                y_splits_restricted = {
+                    split_y.restrict_to(subset=x) for split_y in other.split_sets[j]
+                }
+                if not set(self.split_sets[i]).issubset(y_splits_restricted):
+                    raise MyCheckException()
+        except MyCheckException:
+            return False
+        # (iii)
+        try:
+            for j in range(len(y_parts)):
+                xs_in_y = [x for i, x in enumerate(x_parts) if cover[i] == j]
+                for x1, x2 in it.combinations(xs_in_y, r=2):
+                    sep_sp = [sp for sp in other.split_sets[j] if sp.separates(x1, x2)]
+                    if not sep_sp:
+                        raise MyCheckException()
+        except MyCheckException:
+            return False
+        return True
+
+    def __ge__(self, other):
+        """Check if ``self`` is greater than or equal to ``other``.
+
+        Parameters
+        ----------
+        other : Topology
+            The other topology.
+
+        Returns
+        -------
+        is_greater_than_or_equal : bool
+            Return ``True`` if ``self`` is greater or equal to ``other``, else
+            ``False``.
+        """
+        return other <= self
+
+    def __lt__(self, other):
+        """Check if ``self`` is less than ``other``.
+
+        Parameters
+        ----------
+        other : Topology
+            The other topology.
+
+        Returns
+        -------
+        is_less_than : bool
+            Return ``True`` if ``self`` less than ``other``, else ``False``.
+        """
+        return self <= other and self != other
+
+    def __gt__(self, other):
+        """Check if ``self`` is strictly greater than ``other``.
+
+        Parameters
+        ----------
+        other : Topology
+            The other topology.
+
+        Returns
+        -------
+        is_greater_than : bool
+            Return ``True`` if this topology is greater than the other, else ``False``.
+        """
+        return other < self
+
+    def __hash__(self):
+        """Compute the hash of a topology.
+
+        Note that this hash simply uses the hash function for tuples.
+
+        Returns
+        -------
+        hash_of_topology : int
+            Return the hash of the topology.
+        """
+        return hash((self.n, self.partition, self.split_sets))
+
+
+class Wald(Point):
+    r"""A class for wälder, that are phylogenetic forests, elements of the Wald Space.
+
+    Parameters
```

Comment:
Better naming?

e.g. `self.edge_weights` instead of `self.x`?

`st` is hard to grasp also.

Reply:
named x now weights and st now top as the attribute and topology as the parameter one has to pass the __init__.